### PR TITLE
Add Datasets Page

### PR DESCRIPTION
**Pull request by maciejdrozdowski**
- Adjust GlobalContext state -> from now on it will return [state, dispatch] object
- Move GlobalContext actions definitions to globalActions file
- Add Body of Datasets Page
- Add Datasets related requests
- Add common > components folder (reusable components for all pages)
- Add material-ui-flat-pagination library to handle pagination
- Add Formik & Yup libaries to handle Form state

### CODE REVIEW

File: client/src/Datasets/datasets.api.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { Dataset } from '../common/types';
+import Api from '../common/api';
+
+const baseURL = '/dataset';
+
+export const getAll = async (page: number, limit: number) => {
+    const url = `${baseURL}/data?page=${page}&limit=${limit}`;
```
Comment (wojciech-bilicki) [inline client/src/Datasets/datasets.api.tsx]: we should send the URL query parameters using params dictionary in axis

File: client/src/Datasets/datasets.api.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    const response = await Api.get<Array<Dataset>>(url);
+    return response;
+};
+
+export const create = async (name: string, categories?: Array<string>) => {
+    const url = `${baseURL}/?name=${name}`;
```
Comment (wojciech-bilicki) [inline client/src/Datasets/datasets.api.tsx]: ditto

File: client/src/Datasets/datasets.hooks.ts [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        navigation.navigate(`/dataset/${dataset.id}`);
+    };
+    return {
+        openDetails: openDetails,
+    };
+};
+
+const useList = (itemsPerPage: number): ListState => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const [_globalState, dispatch] = useGlobalContext();
```
Comment (wojciech-bilicki) [inline client/src/Datasets/datasets.hooks.ts]: you should be able to replace the `_globalState` with just `_` and eslint is going to play nicely

File: client/src/common/components/Formik/TextField.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import MuiTextField from '@material-ui/core/TextField';
+import { FormikProps } from 'formik';
+
+const TextField: React.FC<{
+    name: string;
+    label: string;
+    formik: FormikProps<any>;
```
Comment (wojciech-bilicki) [inline client/src/common/components/Formik/TextField.tsx]: We are we using `any` type here?

**Review by wojciech-bilicki — CHANGES_REQUESTED** [review]

**Review by wojciech-bilicki — APPROVED** [review]

**Review by wiola-dreszler — APPROVED** [review]

**Pull request merged**